controllers/utils/conditions.go

```diff
@@ -240,3 +240,45 @@ func GetPreviousStage(stage lcav1alpha1.ImageBasedUpgradeStage) lcav1alpha1.Imag
 	}
 	return ""
 }
+
+// SetUpgradeStatusFailed updates the upgrade status to failed with message
```

Comment:
nit: Should we include the call to updateStatus in these functions as well? These just set the condition in-memory, I believe, with a call to r.updateStatus at the end of the reconciler. Maybe the updateStatus should be moved here as well (as UpdateUpgradeStatus, or UpdateIBUStatus), and called from these functions so that they're updating in-memory and in-db?

Doesn't have to be in this PR. It can be a discussion point and a later PR